autopilot/contractor.go

```diff
@@ -347,13 +347,22 @@ func (c *contractor) performContractMaintenance(ctx context.Context, w Worker) (
 	// check if we need to form contracts and add them to the contract set
 	var formed []types.FileContractID
 	if uint64(len(updatedSet)) < threshold {
-		formed, err = c.runContractFormations(ctx, w, candidates, usedHosts, unusableHosts, state.cfg.Contracts.Amount-uint64(len(updatedSet)), &remaining)
+		// no need to try and form contracts if wallet is completely empty
+		wallet, err := c.ap.bus.Wallet(ctx)
 		if err != nil {
-			c.logger.Errorf("failed to form contracts, err: %v", err) // continue
+			c.logger.Errorf("failed to fetch wallet, err: %v", err)
+			return false, err
+		} else if wallet.Confirmed.IsZero() {
```

Comment:
Can we not form contracts with unconfirmed txns?

Reply:
I'm no wallet expert but I'm pretty sure the way it is now that is not possible. I always manually verified it and the wallet keeps indicating it doesn't have enough funds to fund the transaction. I never got around to adding to `parentTxn` thing to the wallet, mostly because I don't really know how that works. Could take a peek at how siad did that.